lib/src/api/mod.rs

```diff
@@ -1,34 +1,129 @@
 //! Functionality for connecting to local and remote databases
 
-pub mod engine;
-pub mod err;
-#[cfg(feature = "protocol-http")]
-pub mod headers;
-pub mod method;
-pub mod opt;
-
-mod conn;
-
-pub use method::query::Response;
 use method::BoxFuture;
-use semver::Version;
-use tokio::sync::watch;
-
-use crate::api::conn::Router;
-use crate::api::err::Error;
-use crate::api::opt::Endpoint;
 use semver::BuildMetadata;
+use semver::Version;
 use semver::VersionReq;
 use std::fmt;
 use std::fmt::Debug;
 use std::future::IntoFuture;
 use std::marker::PhantomData;
 use std::sync::Arc;
 use std::sync::OnceLock;
+use tokio::sync::watch;
+
+macro_rules! transparent_wrapper{
+	(
+		$(#[$m:meta])*
+		$vis:vis struct $name:ident($field_vis:vis $inner:ty)
+	) => {
+		$(#[$m])*
+		#[repr(transparent)]
+		$vis struct $name($field_vis $inner);
+
+		impl $name{
+			#[doc(hidden)]
+			#[allow(dead_code)]
+			pub fn from_inner(inner: $inner) -> Self{
+				$name(inner)
+			}
+
+			#[doc(hidden)]
+			#[allow(dead_code)]
+			pub fn from_inner_ref(inner: &$inner) -> &Self{
+				unsafe{
```

Comment:
Is unsafe code necessary here? We have avoided unsafe code from the SDK thus far. I think that's a good thing. If we really need to use unsafe blocks, they should be properly documented to justify why they are safe.

Reply:
I can add a safety comment. 
It is mostly here to allow for example `&CoreValue` to be changed into `&Value` in Object::get and other places where you need to wrap a reference. I could pull in bytemuck to not have the unsafe there but I didn't think that was necessary.